pkg/classroom/http.go

```diff
@@ -44,6 +64,18 @@ func ListAcceptedAssignments(client api.RESTClient, assignmentID int, page int,
 		return AcceptedAssignmentList{}, nil
 	}
 
+	//keep calling getAssignmentList until we get them all
+	var nextList []AcceptedAssignment
+	for hasNext := true; hasNext; {
+		page += 1
+		nextList, err = getAssignmentList(client, assignmentID, page, perPage)
```

Comment:
It would be nice if we could run these calls in parallel using [go routines](https://go.dev/tour/concurrency/1) and [channels](https://go.dev/tour/concurrency/2) to wait on errors/responses. 

However, I also realized that `hasNext` relies on the response of the previous call. Maybe we can find a way to optimize this call in a second iteration.

Reply:
@zrdaley I agree that would be cool! So I updated the code to use go routines and channels. Each channel will handle one page of data :) commit: f72155b23d5a307dc59ed8c958b6ef1ac78ade45

Any feedback regarding my usage of channels and go routines would be much appreciated :)

Reply:
Ack! I just saw that this pull request was already merged :) I can open another pull request for the channel updates.